src/main.rs

```diff
@@ -115,34 +130,24 @@ fn get_token_value(client: &mut MPDClient, song: &Song, token: &str) -> String {
 }
 
 fn main() {
-    let config = load_config().unwrap().parse::<Value>().unwrap();
-
-    let app_id = config["id"].as_integer().unwrap() as u64;
-    let hosts: Vec<String> = config["hosts"]
-        .as_array()
-        .unwrap()
-        .iter()
-        .map(|val| val.as_str().unwrap().to_string())
-        .collect();
-
-    let format_options = config.get("format");
-
-    let details_format = match format_options {
-        Some(options) => options.as_table().unwrap()["details"]
-            .as_str()
-            .unwrap_or(DETAILS_FORMAT),
+    let config: Config = toml::from_str(load_config().unwrap().as_str()).unwrap();
+
+    let hosts = &config.hosts;
+
+    let format_options = config.format;
+
+    let details_format = match format_options.as_ref() {
+        Some(options) => options.details.as_str(),
         None => DETAILS_FORMAT,
     };
```

Comment:
This is mostly about taste but you could change this into
```rust
let details_format = format_options.as_ref().map(|s| s.as_str()).unwrap_or(DETAILS_FORMAT);
```

Reply:
You bloody wrote that code in the first place. I'm going to keep is as-is for now because I think it's slightly more readable.